README.md

```diff
@@ -4,6 +4,34 @@
 
 This gem is an implementation of the [language server protocol specification](https://microsoft.github.io/language-server-protocol/) for Ruby, used to improve editor features.
 
+# Overview
+
+The intention of Ruby LSP is to provide a fast, accurate, and robust coding environment for Ruby developers, especially for large codebases.
+The intention of Ruby LSP is to provide a fast, robust and feature-rich coding environment for Ruby developers.
+
+It's part of a [wider Shopify goal](https://github.com/Shopify/vscode-shopify-ruby) to provide a state-of-the-art experience to Ruby developers using modern standards for cross-editor features, documentation and debugging.
+
+It provides many features, including:
+
+* Syntax highlighting
+* Linting and formatting
+* Code folding
+* Selection ranges
+
+It does not perform typechecking, so its features are implemented on a best-effort basis, aiming to be as accurate as possible.
+
+Planned future features include:
+
+* Auto-completion and navigation ("Go To Definition") ([prototype](https://github.com/Shopify/ruby-lsp/pull/429))
+* Support for plug-ins to extend behavior
+
+ruby-lsp does not perform any type-checking or provide any type-related assistance, but it can be used alongside [Sorbet](https://github.com/sorbet/sorbet)'s LSP server.
```

Comment:
```suggestion
The Ruby LSP does not perform any type-checking or provide any type-related assistance, but it can be used alongside [Sorbet](https://github.com/sorbet/sorbet)'s LSP server.
```